### PR TITLE
feat(evm): Update SDK to Support New ERC20Handler with Optional Contract Call

### PR DESCRIPTION
**Pull request by saadjhk**
## Implementation details
With the introduction of the new `ERC20Handler`, the SDK needs to be updated to handle the enhanced functionality that supports both ERC20 token transfers and optional contract calls in a single transaction. [New ERC20Handler PR](https://github.com/sygmaprotocol/sygma-solidity/pull/266)

   - Modify the SDK to work with the updated `ERC20Handler` by allowing it to handle additional data that encodes contract calls alongside ERC20 transfers.
   - Maintain compatibility with existing handlers and workflows that do not involve contract calls, ensuring that the SDK remains functional for all users.

## Testing details
- **Unit Tests**: Develop unit tests that cover scenarios where both ERC20 transfers and contract calls are involved.
- **Compatibility Tests**: Ensure that existing functionality remains unaffected by the changes.
- **Error Handling**: Test how the SDK handles invalid or failed contract calls within the transaction flow.

## Acceptance Criteria
- [ ] The SDK supports interactions with the new `ERC20Handler`, allowing for both ERC20 transfers and optional contract calls within the same transaction.
- [ ] The SDK maintains backward compatibility and continues to function as expected with existing handlers.
- [ ] All new functionality is covered by tests, ensuring reliable and consistent behavior.

### CODE REVIEW

**Comment by lastperson** [comment]
Overall correct, though on the contracts side optionalGas if included must be accompanied by the optionalMessage, otherwise contract call would not happen. So allowing users to only include gas might create a false impression that it affects anything.

**Review by Lykhoyda — APPROVED** [review]

File: examples/evm-to-evm-fungible-transfer/src/transfer.ts [hunk 1]
```diff
@@ -37,14 +38,15 @@ export async function erc20Transfer(): Promise<void> {
...
+  const params: EvmFungibleTransferRequest = {
     source: SEPOLIA_CHAIN_ID,
     destination: AMOY_CHAIN_ID,
     sourceNetworkProvider: web3Provider as unknown as Eip1193Provider,
     resource: RESOURCE_ID,
-    amount: BigInt(1) * BigInt(1e18),
-    recipientAddress: destinationAddress,
-    sourceAddress,
+    amount: BigInt(1) * BigInt(1e6),
+    destinationAddress: destinationAddress, // this needs to be AddressZero to call DefaultMessageReceiver on dest
```
Comment (wainola) [inline examples/evm-to-evm-fungible-transfer/src/transfer.ts:47]: does this comment still makes sense? I guess we can use @remarks here

File: packages/evm/src/utils/depositFn.ts [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+export const executeDeposit = async (
+  domainId: string,
+  resourceId: string,
+  depositData: string,
+  feeData: EvmFee,
+  bridgeInstance: Bridge,
+  overrides?: ethers.PayableOverrides,
+): Promise<PopulatedTransaction> => {
+  const transactionSettings = {
+    // * "twap" and "basic" both deduct in native currency
```
Comment (wainola) [inline packages/evm/src/utils/depositFn.ts:31]: can we use @remarks here from TS docs?

**Review by wainola — DISMISSED** [review]
Left two comments

**Review by Lykhoyda — APPROVED** [review]

**Review by wainola — APPROVED** [review]

**Pull request merged**